src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -525,6 +526,65 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
           // case KeyManager.Ledger:
           // const ledgerWallet = await state.adapters.get(KeyManager.Ledger)?.[0].pairDevice()
           // return ledgerWallet
+          case KeyManager.Ledger: {
+            const featureFlags = preferences.selectors.selectFeatureFlags(store.getState())
+            const isLedgerReadOnlyEnabled = featureFlags.LedgerReadOnly
+
+            // This flag ensures we restore Ledger as connected in terms of state, i.e
+            // wether or not it is *physically* connected
+            if (isLedgerReadOnlyEnabled) {
+              try {
+                // Get the adapter again in each switch case to narrow down the adapter type
+                const ledgerAdapter = await getAdapter(localWalletType)
+
+                if (ledgerAdapter) {
+                  currentAdapters[localWalletType] = ledgerAdapter
+                  dispatch({ type: WalletActions.SET_ADAPTERS, payload: currentAdapters })
+                  // Fixes issue with wallet `type` being null when the wallet is loaded from state
+                  dispatch({
+                    type: WalletActions.SET_CONNECTOR_TYPE,
+                    payload: { modalType: localWalletType, isMipdProvider: false },
+                  })
+                }
+
+                // Don't call pairDevice() here as it requires user gesture for WebUSB permission
+                // Instead, restore the wallet state in read-only mode
+                const { name, icon } = SUPPORTED_WALLETS[KeyManager.Ledger]
+
+                dispatch({
+                  type: WalletActions.SET_WALLET,
+                  payload: {
+                    wallet: null, // No physical wallet connection yet
+                    name,
+                    icon,
+                    deviceId: localWalletDeviceId,
+                    connectedType: KeyManager.Ledger,
+                  },
+                })
+                dispatch({
+                  type: WalletActions.SET_IS_CONNECTED,
+                  payload: true,
+                })
+
+                // Show the Ledger modal and navigate to connect screen
+                dispatch({
+                  type: WalletActions.SET_INITIAL_ROUTE,
+                  payload: '/ledger/connect',
+                })
```

Comment:
Also new - previously would *sometimes* show up as a byproduct of not having a wallet connected e.g if you are in wallet page, which pops up the wallet connect modal, but wouldn't happen in every route.

Now with the flag on, having a Ledger previously connected will only pop this guy, automagically routed to ledger connect route, with the option to connect as read-only, or pair maybe existing/new device as before